quartz/plugins/transformers/roam.ts

```diff
@@ -0,0 +1,132 @@
+import { QuartzTransformerPlugin } from "../types"
+
+export interface Options {
+  orComponent: boolean
+  TODOComponent: boolean
+  DONEComponent: boolean
+  videoComponent: boolean
+  audioComponent: boolean
+  pdfComponent: boolean
+  blockquoteComponent: boolean
+  tableComponent: boolean
+  attributeComponent: boolean
+}
+
+const defaultOptions: Options = {
+  orComponent: true,
+  TODOComponent: true,
+  DONEComponent: true,
+  videoComponent: true,
+  audioComponent: true,
+  pdfComponent: true,
+  blockquoteComponent: true,
+  tableComponent: true,
+  attributeComponent: true,
+}
+
+const orRegex = new RegExp(/{{or:(.*?)}}/, "g")
+const TODORegex = new RegExp(/{{.*?\bTODO\b.*?}}/, "g")
+const DONERegex = new RegExp(/{{.*?\bDONE\b.*?}}/, "g")
+const videoRegex = new RegExp(/{{.*?\bvideo\b.*?\:(.*?)}}/, "g")
+const youtubeRegex = new RegExp(/{{.*?\bvideo\b.*?(\bhttp.*?\byoutu.*?)watch\?v=(.*?)}}/, "g")
+
+const audioRegex = new RegExp(/{{.*?\baudio\b.*?\:(.*?)}}/, "g")
+const pdfRegex = new RegExp(/{{.*?\bpdf\b.*?\:(.*?)}}/, "g")
+const blockquoteRegex = new RegExp(/\[\[>\]\]/, "g")
+const roamHighlightRegex = new RegExp(/\^\^(.+)\^\^/, "g")
+const roamItalicRegex = new RegExp(/__(.+)__/, "g")
+const tableRegex = new RegExp(/- {{.*?\btable\b.*?}}/, "g") /* TODO */
+const attributeRegex = new RegExp(/\b\w+(?:\s+\w+)*::/, "g") /* TODO */
+
+export const RoamFlavoredMarkdown: QuartzTransformerPlugin<Partial<Options> | undefined> = (
+  userOpts,
+) => {
+  const opts = { ...defaultOptions, ...userOpts }
+
+  return {
+    name: "RoamFlavoredMarkdown",
+    textTransform(_ctx, src) {
+      if (opts.orComponent) {
+        src = src.toString()
+        src = src.replaceAll(orRegex, (value, ...capture) => {
+          const [match, text] = capture
+          const options = match.split("|")
+          const id = Math.random().toString(36).substring(2, 15) // Generate a unique ID
```

Comment:
What is this used for?

Reply:
In Roam there is a component `{{ or: Option 1 | Option 2 | Option 3 }}` that creates a dropdown select. This can be nice while writing something unpolished as you can have multiple ways of wording something
<img width="245" alt="image" src="https://github.com/jackyzha0/quartz/assets/4028391/ba94cd15-318b-474d-ad29-b288e0e20614">

Reply:
I meant this part:

```js
const id = Math.random().toString(36).substring(2, 15) // Generate a unique ID
```

Reply:
Oh, looks like old code. In a previous version I used a custom carousel instead of an html dropdown. I still think that's a more elegant solution but it's way more complex so I simplified it to a normal html select for this pr. I'll remove the line
![Kapture 2024-03-11 at 08 18 03](https://github.com/jackyzha0/quartz/assets/4028391/12939d43-bbf3-45e0-95c6-4327b55ee6b2)
e